### PR TITLE
GCI CSY2002 message

### PR DESCRIPTION
**Pull request by sycDev**

### CODE REVIEW

File: src/main.go [hunk 1]
```diff
@@ -1,27 +1,26 @@
 package main
 
 import (
-	"fmt"
 	"github.com/gin-gonic/gin"
 )
 
-func HomePage(c *gin.Context) {
+func Hello(c *gin.Context) {
```
Comment (suthagar23) [inline src/main.go]: Better to add method comments

File: src/main.go [hunk 2]
```diff
@@ -1,27 +1,26 @@
...
 	c.JSON(200, gin.H{
-		"message": "Hello World",
+		"message": "Hello World!",
 	})
 }
 
-func PostHomePage(c *gin.Context) {
+func Greet(c *gin.Context) {
```
Comment (suthagar23) [inline src/main.go]: can you refactor this to use the proper meaningful variable names?

File: src/main.go (hunk 1 above)
Comment (f4ww4z) [reply src/main.go]: @sycDev Did you see this?

File: src/main.go [hunk 3]
```diff
@@ -1,27 +1,28 @@
...
+func Hello(c *gin.Context) {
+	// Return message "Hello World!" in JSON format
 	c.JSON(200, gin.H{
-		"message": "Hello World",
+		"message": "Hello World!",
 	})
 }
 
-func PostHomePage(c *gin.Context) {
+func HelloName(c *gin.Context) {
```
Comment (f4ww4z) [inline src/main.go]: Add a method comment.

**Comment by f4ww4z** [comment]
@sycDev  Have you googled how to rebase? [First search result.](https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git) Also, fix your PR title by describing the changes in the `message` part.

File: src/main.go [hunk 4]
```diff
@@ -1,27 +1,28 @@
 package main
 
 import (
-	"fmt"
 	"github.com/gin-gonic/gin"
 )
 
-func HomePage(c *gin.Context) {
+func Hello(c *gin.Context) {
+	// Return message "Hello World!" in JSON format
```
Comment (f4ww4z) [inline src/main.go]: @sycDev This should be positioned above the method. [See this tutorial](https://blog.golang.org/godoc-documenting-go-code).

File: src/main.go [hunk 5]
```diff
@@ -1,27 +1,28 @@
...
 	c.JSON(200, gin.H{
-		"message": "Hello World",
+		"message": "Hello World!",
 	})
 }
 
-func PostHomePage(c *gin.Context) {
+func HelloName(c *gin.Context) {
+	// Passing parameter :name  to output message "Hello :name!" in JSON format
```
Comment (f4ww4z) [inline src/main.go]: https://blog.golang.org/godoc-documenting-go-code

File: src/main.go [hunk 6]
```diff
@@ -1,27 +1,28 @@
...
+	name := c.Param("name")
```
Comment (f4ww4z) [inline src/main.go]: Can you make `"name"` a global constant?

File: src/main.go [hunk 7]
```diff
@@ -1,27 +1,30 @@
...
 	c.JSON(200, gin.H{
-		"message": "Hello World",
+		"message": "Hello World!",
 	})
 }
 
-func PostHomePage(c *gin.Context) {
+// Return message "Hello" and name parameter in JSON format
+func HelloName(c *gin.Context) {
+	name := c.Param("name")
```
Comment (f4ww4z) [inline src/main.go]: No, the constant is used for the param. `name := c.Param(Name)`

File: src/main.go [hunk 8]
```diff
@@ -1,27 +1,30 @@
 package main
 
 import (
-	"fmt"
 	"github.com/gin-gonic/gin"
 )
 
-func HomePage(c *gin.Context) {
+var name = ""
```
Comment (f4ww4z) [inline src/main.go]:
```suggestion
const Name = "name"
```

**Pull request merged**